R/aggregate_by_matrix.R

```diff
@@ -0,0 +1,37 @@
+#' @rdname aggregate
+#'
+#' @export
+colMeansMat <- function(x, MAT)
+  colSumsMat(x, MAT)/colSums(MAT)
+
+#' @rdname aggregate
+#'
+#' @export
+colSumsMat <- function(x, MAT)
+  t(MAT) %*% x
+
+
+##' @param MAT An adjacency matrix that defines what features with
+##'     `nrow(MAT) == nrow(x)`
+##'
+##' @return [aggregate_by_matrix()] returns a `matrix` (or `Matrix`)
+##'     of dimensions `ncol(MAT)` and `ncol(x), with `dimnames` equal
+##'     to `colnames(x)` and `rownames(MAT)`.
+##'
+##' @rdname aggregate
+##'
+##' @export
+aggregate_by_matrix <- function(x, MAT, FUN, ...) {
+    if (!inherits(x, "matrix"))
```

Comment:
Wouldn't be `is.matrix` sufficient here?

Reply:
I think they would be equivalent here. I am just used to `is()` or `inherits()`. I don't have a strong opinion here, so happy to change.